examples/closure-app/silabs/src/ClosureManager.cpp

```diff
@@ -230,19 +237,257 @@ ClosureManager::OnMoveToCommand(const chip::Optional<chip::app::Clusters::Closur
+            ep1CurrentState.Value().latch.SetValue(DataModel::MakeNullable(false));
+            instance.ep1.GetLogic().SetOverallCurrentState(ep1CurrentState);
+            ep2CurrentState.Value().latch.SetValue(false);
+            instance.ep2.GetLogic().SetCurrentState(ep2CurrentState);
+            ep3CurrentState.Value().latch.SetValue(false);
+            instance.ep3.GetLogic().SetCurrentState(ep3CurrentState);
+            ChipLogProgress(AppServer, "Unlatched action completed");
+        }
+    }
+
+    // Once Closure is unlatched, we can proceed with the motion action for endpoints 2 and 3.
+    Optional<chip::Percent100ths> ep2NextPosition = NullOptional;
+    Optional<chip::Percent100ths> ep3NextPosition = NullOptional;
+
+    bool isEndPoint2ProgressPossible = false;
+    bool isEndPoint3ProgressPossible = false;
+
+    // Get the Next Current State to be set for the endpoint 2, if target postion is not reached.
+    if (GetPanelNextPosition(ep2CurrentState.Value(), ep2TargetState.Value(), ep2NextPosition))
+    {
+        VerifyOrReturn(ep2NextPosition.HasValue(),
+                       ChipLogError(AppServer, "Failed to get next position for Endpoint 2"));
+        ep2CurrentState.Value().position.SetValue(ep2NextPosition.Value());
+        instance.ep2.GetLogic().SetCurrentState(ep2CurrentState);
+        isEndPoint2ProgressPossible = (ep2NextPosition.Value() != ep2TargetState.Value().position.Value());
+        ChipLogProgress(AppServer, "EndPoint 2 Current Position: %d, Target Position: %d",
+                        ep2NextPosition.Value(), ep2TargetState.Value().position.Value());
+    }
+
+    // Get the Next Current State to be set for the endpoint 3, if target postion is not reached.
+    if (GetPanelNextPosition(ep3CurrentState.Value(), ep3TargetState.Value(), ep3NextPosition))
+    {
+        VerifyOrReturn(ep3NextPosition.HasValue(),
+                       ChipLogError(AppServer, "Failed to get next position for Endpoint 3"));
+        ep3CurrentState.Value().position.SetValue(ep3NextPosition.Value());
+        instance.ep3.GetLogic().SetCurrentState(ep3CurrentState);
+        isEndPoint3ProgressPossible = (ep3NextPosition.Value() != ep3TargetState.Value().position.Value());
+        ChipLogProgress(AppServer, "EndPoint 3 Current Position: %d, Target Position: %d",
+                        ep3NextPosition.Value(), ep3TargetState.Value().position.Value());
+    }
+
+    // Check if both endpoints have reached their target positions
+    // If both endpoints have reached their target positions, we can consider the closure motion action as complete.
+    // If either endpoint has not reached its target position, we will continue the motion action
+    // and set the closureTargetReached flag to false.
+    // This will ensure that the closure motion action continues until both endpoints have reached their target positions.
+    bool isProgressPossible = isEndPoint2ProgressPossible && isEndPoint3ProgressPossible;
```

Comment:
I'm not sure this is the correct check - it possible for the panels to have different mouvements to execute and thus not take the same amount of time to execute.
We should continue to schedule motion as long as one panel hans't reached its target.